Vagrantfile

```diff
@@ -6,6 +6,196 @@ $NODEMEM=256
+
+#
+# Bash shell settings: exit on failing commands, unbound variables
+#
+set -o errexit
+set -o nounset
+set -o pipefail
+
+#
+# Variables
+#
+# Color definitions
+readonly reset='\e[0m'
+readonly red='\e[0;31m'
+readonly yellow='\e[0;33m'
+readonly cyan='\e[0;36m'
+
+readonly workshop_user=vagrant
+
+readonly workshop_key="/home/${workshop_user}/.ssh/id_rsa"
+readonly workshop_inventory="/home/${workshop_user}/hosts.ini"
+readonly workshop_ansible_config="/home/${workshop_user}/.ansible.cfg"
+
+readonly workshop_git_repo="https://github.com/ansible/lightbulb.git"
+
+#
+# Functions
+#
+
+main() {
+  info "Bootstrapping Ansible on host ${HOSTNAME}."
+
+  ensure_ansible_installed
+  install_vagrant_private_key
+  configure_ansible
+  install_inventory
+  install_lightbulb
+
+}
+
+# Ansible installation
+ensure_ansible_installed() {
+  if ! is_ansible_installed; then
+    distro=$(get_linux_distribution)
+    "install_ansible_${distro}" || die "Distribution ${distro} is not supported"
+  fi
+
+  info "Ansible version"
+  ansible --version
+}
+
+is_ansible_installed() {
+  which ansible-playbook > /dev/null 2>&1
+}
+
+# Print the Linux distribution
+get_linux_distribution() {
+
+  if [ -f '/etc/redhat-release' ]; then
+
+    # RedHat-based distributions
+    cut --fields=1 --delimiter=' ' '/etc/redhat-release' \
+      | tr "[:upper:]" "[:lower:]"
+
+  elif [ -f '/etc/lsb-release' ]; then
+
+    # Debian-based distributions
+    grep DISTRIB_ID '/etc/lsb-release' \
+      | cut --delimiter='=' --fields=2 \
+      | tr "[:upper:]" "[:lower:]"
+
+  fi
+}
+
+# Install Ansible on a Fedora system.
+# The version in the repos is fairly current, so we'll install that
+install_ansible_fedora() {
+  info "Fedora: installing Ansible from distribution repositories"
+  dnf -y install git
+}
+
+# Install Ansible on a CentOS system from EPEL
+install_ansible_centos() {
+  info "CentOS: installing Ansible from the EPEL repository"
+  yum -y install epel-release
+  yum -y install git
+}
+
+# Install Ansible on a recent Ubuntu distribution, from the PPA
+install_ansible_ubuntu() {
+  info "Ubuntu: installing Ansible from PPA"
+  # Remark: on older Ubuntu versions, it's python-software-properties
+  apt-get -y install software-properties-common
+  apt-add-repository -y ppa:ansible/ansible
+  apt-get -y update
+  apt-get -y install git
+}
+
+# Install the Vagrant private key to SSH to the other nodes
+install_vagrant_private_key() {
+  curl --output "${workshop_key}" \
+    https://raw.githubusercontent.com/hashicorp/vagrant/master/keys/vagrant
+  chown "${workshop_user}:${workshop_user}" "${workshop_key}"
+  chmod 0600 "${workshop_key}"
+}
+
+# Install an inventory file
+install_inventory() {
+  info "Installing inventory file ${workshop_inventory}"
+
+  cat > "${workshop_inventory}" << _EOF_
+[control]
+ansible ansible_host=10.42.0.2 ansible_connection=local
+
+[web]
+node-1 ansible_host=10.42.0.6
```

Comment:
I'd add integration with `vagrant-hostmanager` plugin instead of hardcoding IPs

Reply:
Thanks for the suggestion. These fixed static IP addresses are used throughout the content that we want to keep it simple and static like this to demonstrate the static inventory features of Ansible.

Reply:
Yes, but I had to track all occurrences of them once, when I had some issues with conflicting networking. I'd not want to do this again as patching vagrantfile makes it harder to sync with upstream etc. IMHO this should be easily configurable, but now it's painful sometimes, as opposed to "just works" approach.